src/lib.ts

```diff
@@ -110,6 +110,19 @@ function parseXML(content: string) {
   return xml;
 }
 
+/**
+  * Determine the length of signature that will be generated when signing with
+  * the given private key using the SHA1 digest algorithm.
+  */
+function signatureLength(privateKey: string) {
+  // determine signature size by signing some dummy data
+  let signer = createSign('SHA1');
+  signer.update('dummy data');
+  let signature: Buffer = <any>signer.sign(privateKey,
+    undefined /* return a Buffer */);
+  return signature.length;
```

Comment:
Personally, I would rather reuse code. So rename this function into `sign(privateKey, data)` and do `sign(key, '').length` to calculate signature size (note that an empty string will do here because what's actually being signed is the SHA1 digest and that one has a constant size). You could then call `heapWriter.write(sign(key, compressedTOC))` when you need the actual signature.